src/index.ts

```diff
@@ -74,6 +74,8 @@ function activateFileBrowser(app: JupyterLab, manager: IDocumentManager, factory
   // Fetch the initial state of the settings.
   Promise.all([settingRegistry.load(PLUGIN_ID), app.restored])
   .then(([settings]) => {
+    const gitHubBaseUrl = settings.get('gitHubBaseUrl').composite as string;
+    gitHubBrowser.gitHubBaseUrl = gitHubBaseUrl;
```

Comment:
Since this happens at instantiation time you need to refresh the page to have the updated settings take effect - not sure if that's an issue? I don't personally mind that but I suppose it would be slicker / more user friendly if it was set dynamically.

Reply:
You are absolutely right. I skipped dynamically updating the settings here, since the `defaultRepo` setting only applied upon page load. But now, it makes sense to update them. The settings object has a `changed` signal which you can listen to, and update the  `_gitHubBaseUrl` in response to changes.

Several of the core packages show this pattern. The one in `docmanager-extension` is particularly simple example.